### PR TITLE
Pulling code for GRD-75254, GRD-73301, and GRD-73302

### PR DESCRIPTION
**Pull request by chirag-ibm**

### CODE REVIEW

**Review by ofer-haim — APPROVED** [review]

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 1]
```diff
@@ -1,264 +1,192 @@
...
-    
-			4. Provide 'Server Name'. Keep 'Data source' as none ,select appropriate location and set 'Version=11'.
-    
-			5. Keep compute + storage as 'General Purpose, 4 vCore(s), 100 GB'.
-    
-			6. Provide server admin username and password.
-    
+2. Click search bar.
+3. Search for azure database for postgreSQL servers.
+4. Click on 'Create' button.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:25]:
```suggestion
4. Click **Create**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 2]
```diff
@@ -1,264 +1,192 @@
...
+5. Select resource type.(here we use single server.)
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:26]:
```suggestion
5. Select a resource type. (Here we selected single server.)
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 3]
```diff
@@ -1,264 +1,192 @@
...
+6. Select Single server and click on create button and fill details in basic option.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:27]:
```suggestion
6. Select **Single server** > **create** and fill in the details for the **basic** option.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 4]
```diff
@@ -1,264 +1,192 @@
...
+7. In Basic option:
+   1. Select your subscription name.
+   2. Select existing resource group or create new one.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:30]:
```suggestion
   2. Select an existing resource group or create a new one.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 5]
```diff
@@ -1,264 +1,192 @@
...
+   3. For new resource group click on create button  and enter name for resource group.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:31]:
```suggestion
   3. For a new resource group, click **create**  and enter a name for **resource group**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 6]
```diff
@@ -1,264 +1,192 @@
...
+   4. Provide 'Server Name'. Keep 'Data source' as none ,select appropriate location and set 'Version=11'.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:32]:
```suggestion
   4. Fille in the **Server Name** field. Keep **Data source** as **none** ,select the appropriate location, and set **Version=11**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 7]
```diff
@@ -1,264 +1,192 @@
...
+   5. Keep compute + storage as 'General Purpose, 4 vCore(s), 100 GB'.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:33]:
```suggestion
   5. Keep **compute + storage** as **General Purpose, 4 vCore(s), 100 GB**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 8]
```diff
@@ -1,264 +1,192 @@
...
+   6. Provide server admin username and password.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:34]:
```suggestion
   6. Provide a server admin username and password.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 9]
```diff
@@ -1,264 +1,192 @@
...
-    
-12.	After successful creation of resource group, select resource button.
-    	
-13.	From settings,select connection security.
-
-14.	In firewall rules,select allow access to azure services to 'Yes'.
-
-15. Click on add current client IP address, and then click on Save.
-## Enabling Auditing
+9.	Verify the configuration and then click on 'Create' Button.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:36]:
```suggestion
9.	Verify the configuration and then click **Create**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 10]
```diff
@@ -1,264 +1,192 @@
...
+10.	After successful creation of resource group, select resource button.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:37]:
```suggestion
10.	After you successfully create a resource group, select **resource**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 11]
```diff
@@ -1,264 +1,192 @@
...
+11.	From settings,select connection security.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:38]:
```suggestion
11.	From **settings**, select **connection security**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 12]
```diff
@@ -1,264 +1,192 @@
...
+12.	In firewall rules,select allow access to azure services to 'Yes'.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:39]:
```suggestion
12.	In **firewall rules**, select **Yes** for **allow access to azure services**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 13]
```diff
@@ -1,264 +1,192 @@
...
+13. Click on add current client IP address, and then click on Save.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:40]:
```suggestion
13. Click **add current client IP address**, and then click **Save**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 14]
```diff
@@ -1,264 +1,192 @@
...
-       •	log_line_prefix = specify as per requirement but should include timestamp,client ip,client port,database 
-   username,database name,process id application name,sql state. Refer from this link https://www.postgresql.org/docs/current/runtime-config-logging.html#GUC-LOG-LINE-PREFIX 
-							(eg:- %t:%r:%u@%d:[%p]:%a:%e)
-       
-       •	pgaudit.log = DDL,FUNCTION,READ,WRITE,ROLE
-   
-       •	pgaudit.log_catalog = off
-			
-       •	pgaudit.log_client = off
+      1. Under Settings, select server parameters and set server parameters as follows:
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:50]:
```suggestion
      1. In **Settings**, select **server parameters** and set the server parameters as follows:
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 15]
```diff
@@ -1,264 +1,192 @@
...
+         * log_error_verbosity = VERBOSE 
+         * log_line_prefix = specify as per requirement but should include 
+         timestamp, client ip, client port, database username, database name, process id, application name,sql state. 
+         Refer from this link https://www.postgresql.org/docs/current/runtime-config-logging.html#GUC-LOG-LINE-PREFIX
+         (eg:- %t:%r:%u@%d:[%p]:%a:%e)
+         * pgaudit.log = DDL,FUNCTION,READ,WRITE,ROLE 
+         * pgaudit.log_catalog = off 
+         * pgaudit.log_client = off 
+         * pgaudit.log_parameter = off 
+6. Click save.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:64]:
```suggestion
6. Click **save**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 16]
```diff
@@ -1,264 +1,192 @@
...
-   	
-				5. For gmachine , We have to use these consumer group name in input section of configuration file as follows:
-   	
-					1. On one machine in input section of configuration file in consumer group field give name as `$Default`.
-   	
-					2. On other machine in input section of configuration file in consumer group field give name of other consumer group.
-   	
-					3. Keep all other configurations as it is.
-		
+1. Search Storage Accounts in the search bar of the Azure portal.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:71]:
```suggestion
1. Search for Storage Accounts in the search bar of the Azure portal.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 17]
```diff
@@ -1,264 +1,192 @@
...
+
+5. To create an event hub:
+    1. Go to the Event Hubs Namespace page.
+    2. Click **+ Event Hub** to add event hub.
+    3. Enter a unique name for the event hub, then select **create**.
+
+6. Connection string for a eventhub:
+    1. In the list of event hubs, select your event hub.
+    2. On the Event Hubs instance page, go to **settings**  >  **shared access policies** > **Add**.
+    3. In shared access policies click on add button from top.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:101]:
```suggestion
    3. In shared access policies click **add**.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 18]
```diff
@@ -1,264 +1,192 @@
...
+    4. Provide the name to policy and choose **manage** from the permissions and create the policy.
+    5. Take the value of the primary key - connection string from the policy (this is required for an input plug-in).
+    6. The value of the Primary Key - Connection String will be used in the parameter **event_hub_connections** in the input of the plugin.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:104]:
```suggestion
    6. The value of the primary key - connection string will be used in the parameter **event_hub_connections** in the input of the plug-in.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 19]
```diff
@@ -1,264 +1,192 @@
...
+
+7. Stream logs to an event hub:
+    1. Go to the server in your Azure portal.
+    2. From **Monitoring**, select **Diagnostics settings** and do either of the following:
+        1. Change existing settings by selecting **Edit setting**.
+        2. Add new settings by selecting **Add diagnostics setting**.
+        3. To add new settings, do following:
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:111]:
```suggestion
        3. To add new settings, do the following:
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 20]
```diff
@@ -1,264 +1,192 @@
...
 
 ## 4. Connecting to Azure postgreSQL Database:
 
 1. Start the psql and provide connection details.
-
-2. Enter the 'server name' (from overview window of server in azure portal)
-
-3. Enter 'database name' as 'postgres'.
-
+2. Enter the **server name** (from overview window of server in azure portal)
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:153]:
```suggestion
2. Enter the **server name** (from the overview window of the server in the Azure portal).
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 21]
```diff
@@ -1,264 +1,192 @@
...
+3. Enter the database name as 'postgres'.
 4. Enter 'port' as 5432.
-
- 5. Provide the 'username' (you will get it from azure portal,click on overview and copy admin username) and ‘password’ that we had set while creating the database.
-
+5. Provide the 'username' (you will get it from azure portal,click on overview and copy admin username) and ‘password’ that we had set while creating the database.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:156]:
```suggestion
5. Provide the 'username' (you can get it from the Azure portal. Click **overview** and copy the admin username and ‘password’ that you set when you created the database).
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 22]
```diff
@@ -1,264 +1,192 @@
...
 
 #### Limitations
-• The azure postgreSQL plug-in does not support IPV6.
-
-• For sql errors and login failed,sql string is not available.
-
-• For primary or foreign key constraints violation, entry would be added to sql error report as well as full sql report.
-
-• Source program will be seen as blank in report for some clients(here for psql and pgadmin we get value but for visual studio it is blank).
+* The azure postgreSQL plug-in does not support IPV6.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:166]:
```suggestion
* The Azure PostgreSQL plug-in does not support IPV6.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 23]
```diff
@@ -1,264 +1,192 @@
...
+* For sql errors and login failed,sql string is not available.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:167]:
```suggestion
* For SQL errors and failed logins, SQL string is not available.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 24]
```diff
@@ -1,264 +1,192 @@
...
+* For primary or foreign key constraints violation, entry would be added to sql error report as well as full sql report.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:168]:
```suggestion
* For primary or foreign key constraints violations, entry are added to both the SQL error report and the full SQL report.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 25]
```diff
@@ -1,264 +1,192 @@
...
+* Source program will be seen as blank in report for some clients(here for psql and pgadmin we get value but for visual studio it is blank).
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:169]:
```suggestion
* The source program will be seen as blank in the report for some clients (in this case, values appear for psql and pgadmin but visual studio is blank).
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 26]
```diff
@@ -1,264 +1,192 @@
...
 
 ## 5. Configuring the Azure PostgreSQL filters in Guardium
 
 The Guardium universal connector is the Guardium entry point for native audit logs. The Guardium universal connector identifies and parses the received events, and converts them to a standard Guardium format. The output of the Guardium universal connector is forwarded to the Guardium sniffer on the collector, for policy and auditing enforcements. Configure Guardium to read the native audit logs by customizing the azure postgreSQL template.
 
 
 #### Before you begin
+* You must have LFD policy enabled on the collector. The detailed steps can be found in step 4 on [this page](https://www.ibm.com/docs/en/guardium/11.4?topic=dpi-installing-testing-filter-input-plug-in-staging-guardium-system).
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:177]:
```suggestion
* Configure the policies you require. See [policies](/docs/#policies) for more information.
```

Comment (AlexaNahum) [reply filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:177]: Please ONLY work off existing plug-in readmes that are live in GitHub when you are creating new instructions . That way we can ensure that all the information is current and all my previous edits continue to appear.

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 27]
```diff
@@ -1,264 +1,192 @@
...
+* You must have permission for the S-Tap Management role. The admin user includes this role by default. 
+* Download the [azure-postgresql-offline-plugins-7.5.2.zip](https://github.ibm.com/Activity-Insights/universal-connectors/blob/master/filter-plugin/logstash-filter-azure-postgresql-guardium/AzurePostgresqlOverAzureEventHub/azurepostgresql/azure-postgresql-offline-plugins-7.5.2.zip) plug-in.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:179]:
```suggestion
* Download the [azure-postgresql-offline-plugins-7.5.2.zip](https://github.ibm.com/Activity-Insights/universal-connectors/blob/master/filter-plugin/logstash-filter-azure-postgresql-guardium/AzurePostgresqlOverAzureEventHub/azurepostgresql/azure-postgresql-offline-plugins-7.5.2.zip) plug-in. (Do not unzip the offline-package file throughout the procedure). This step is not necessary for Guardium Data Protection v12.0 and later.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 28]
```diff
@@ -1,264 +1,192 @@
...
 #### Procedure : 
 
 1.	On the collector, go to Setup > Tools and Views > Configure Universal Connector.
-
-2.	Enable the universal connector if it is disabled.
-
-3.	Click Upload File and select the offline [azure-postgresql-offline-plugins-7.5.2.zip](https://github.com/IBM/universal-connectors/raw/release-v1.2.0/filter-plugin/logstash-filter-azure-postgresql-guardium/AzurePostgresqlOverAzureEventHub/azurepostgresql/azure-postgresql-offline-plugins-7.5.2.zip) plugin. After it is uploaded,click OK. This step is not necessary for Guardium Data Protection v12.0 and later.
-
+2.	First enable the Universal Guardium connector, if it is disabled already.
+3.	Click Upload File and select the offline [azure-postgresql-offline-plugins-7.5.2.zip](https://github.ibm.com/Activity-Insights/universal-connectors/blob/master/filter-plugin/logstash-filter-azure-postgresql-guardium/AzurePostgresqlOverAzureEventHub/azurepostgresql/azure-postgresql-offline-plugins-7.5.2.zip) plugin. After it is uploaded,click OK.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:186]:
```suggestion
3.	Click Upload File and select the offline [azure-postgresql-offline-plugins-7.5.2.zip](https://github.ibm.com/Activity-Insights/universal-connectors/blob/master/filter-plugin/logstash-filter-azure-postgresql-guardium/AzurePostgresqlOverAzureEventHub/azurepostgresql/azure-postgresql-offline-plugins-7.5.2.zip) plugin. After it is uploaded,click OK. This is not necessary for Guardium Data Protection v12.0 and later.
```

File: filter-plugin/logstash-filter-azure-postgresql-guardium/README.md [hunk 29]
```diff
@@ -1,264 +1,192 @@
...
 
 #### Procedure : 
 
 1.	On the collector, go to Setup > Tools and Views > Configure Universal Connector.
-
-2.	Enable the universal connector if it is disabled.
-
-3.	Click Upload File and select the offline [azure-postgresql-offline-plugins-7.5.2.zip](https://github.com/IBM/universal-connectors/raw/release-v1.2.0/filter-plugin/logstash-filter-azure-postgresql-guardium/AzurePostgresqlOverAzureEventHub/azurepostgresql/azure-postgresql-offline-plugins-7.5.2.zip) plugin. After it is uploaded,click OK. This step is not necessary for Guardium Data Protection v12.0 and later.
-
+2.	First enable the Universal Guardium connector, if it is disabled already.
```
Comment (AlexaNahum) [inline filter-plugin/logstash-filter-azure-postgresql-guardium/README.md:185]:
```suggestion
2.	First enable the universal connector if it is currently disabled.
```

**Review by AlexaNahum — APPROVED** [review]

**Pull request closed without merging**